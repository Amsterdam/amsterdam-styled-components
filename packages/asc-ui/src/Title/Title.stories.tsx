import { storiesOf } from '@storybook/react';
import * as React from 'react';
import Title from './Title';

const stories = storiesOf('Title', module);

stories.add(
    'Normal',
    () => <Title>This is a title</Title>,
    { info: { inline: true } },
);
