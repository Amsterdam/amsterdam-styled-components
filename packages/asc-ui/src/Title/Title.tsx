import { default as styled } from '../custom-theme'
import theme from '../theme'

interface IProps {
  theme: theme
}

const Title = styled.h1`
  margin-top: 20px;
  color: ${(props: IProps) => props.theme.primaryColor};
`

export default Title
