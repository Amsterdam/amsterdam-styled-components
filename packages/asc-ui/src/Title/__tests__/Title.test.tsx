import { mount } from 'enzyme'
import 'jest-styled-components'
import * as React from 'react'
import Title from '../Title'

describe('<Title />', () => {
  it('should render', () => {
    const component = mount(<Title />)
    expect(component).toMatchSnapshot()
  })
})
