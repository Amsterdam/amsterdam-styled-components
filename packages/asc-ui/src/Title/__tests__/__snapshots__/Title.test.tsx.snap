// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> should render 1`] = `
.c0 {
  margin-top: 20px;
}

<styled.h1>
  <StyledComponent
    forwardedClass={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  margin-top: 20px;
  color: ",
            [Function],
            ";
",
          ],
        },
        "displayName": "styled.h1",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "h1",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <h1
      className="c0"
    />
  </StyledComponent>
</styled.h1>
`;
